src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs

```diff
@@ -172,6 +173,14 @@ public static SBMessage AmqpMessageToSBMessage(AmqpMessage amqpMessage)
                 var byteArrayValue = (byte[])amqpMessage.ValueBody.Value;
                 sbMessage = new SBMessage(byteArrayValue);
             }
+            else if (amqpMessage.BodyType == SectionFlag.Data
+                && amqpMessage.DataBody != null
+                && amqpMessage.DataBody.Count() > 0)
+            {
+                var dataBody = amqpMessage.DataBody.FirstOrDefault().Value;
```

Comment:
Would we ever need more thant he first DataBody? If so, what would be the best way to concatenate them?